File: 1.two-sum.js
/*
 * @lc app=leetcode id=1 lang=javascript
 *
 * [1] Two Sum
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    var mp = new Map();
    for(let i =0; i < nums.length; i++) {
        let tmp = target - nums[i];
        
        if (mp.has(tmp)){
            return [mp.get(tmp),i];
        }
        mp.set(nums[i],i);

    }
    
};
// @lc code=end

File: 1.two-sum.py
#
# @lc app=leetcode id=1 lang=python3
#
# [1] Two Sum
#

# @lc code=start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map={}
        for i in range(len(nums)):
            res = target - nums[i]
            if res not in map.keys():
                map[nums[i]] = i
            else:
                return [i,map.get(res)]
        return []      
# @lc code=end
